public/js/core.js

```diff
@@ -0,0 +1,193 @@
+angular.
+module('taskApp', []).
+controller('mainController', ['$scope', '$http', '$filter', 'getTaskFromWebSocket', function($scope, $http, $filter, getTaskFromWebSocket) {
+
+    $scope.formData = {};
+    $scope.currentElement = {};
+    $scope.isSelect = true;
+    $scope.Active = true;
+    $scope.Completed = true;
+    var socket = io();
+    // when landing on the page, get all taskList and show them
+
+    //WEB SOCKET
+    $scope.getTasksIo = function() {
+        console.log("call");
+        var tmp = socket.emit('getTaskListIo');
+        console.log('----');
+        console.log(tmp);
+    };
+
+    $scope.getTaskIo = function(id) {
+        getTaskFromWebSocket.getTaskListIo();
+        $scope.isSelect = false;
+        console.log("call getTaskIo");
+        socket.emit('getTaskIo', id);
+        socket.on('getTaskIo', function(msg) {
+            console.log('----');
+            $scope.currentElement = tmp[0];
+            console.log("a----");
+            console.log($scope.currentElement);
+            $scope.currentElement.priority = $scope.selectOptions[$scope.currentElement.priority - 1];
+        });
+    };
+
+    //REST
+    $http.get('/api/taskList')
```

Comment:
Create special services for all of these requests using ngResource or restangular.